src/commons/editor/Editor.tsx

```diff
@@ -149,13 +149,26 @@ const makeHandleGutterClick =
  * array has empty slots (i.e., array indices which are not defined at all). Breakpoints
  * are represented in the Ace Editor's breakpoints array as key-value pairs where the
  * key is a string representation of the line number it is on, and the value is
- * 'ace_breakpoint'. As such, to get an array of breakpoint line numbers, we need to
- * get the keys of the Ace Editor's breakpoint array/object and parse them as integers.
+ * 'ace_breakpoint'.
+ *
+ * However, because the JSON-serialised representation of empty slots in a JavaScript
+ * array is the same as that of the 'null', we cannot simply parse all keys of the object
```

Comment:
Actually, accessing the value of an empty array slot results in undefined, not null. The fix is still valid I think, but the comment should change to avoid confusion.

Reply:
I was previously accessing the array as an object based on the observation that empty slots in an array don't even exist as key-value pairs in the underlying object. After thinking about it more from your comment though, I realised that since we now have to check the value of elements in the array anyway, it no longer makes sense to think in terms of object semantics when we can just make use of array semantics for a cleaner implementation. This also means that most of the comment can be removed.

Thanks for bringing this up! I'll make the change (for both code and comment).